core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java

```diff
@@ -153,7 +157,7 @@ private static void updateMetrics(RequestLog log, RequestMetrics metrics) {
         metrics.responseLength().record(log.responseLength());
         metrics.totalDuration().record(log.totalDurationNanos(), TimeUnit.NANOSECONDS);
 
-        if (isSuccess(log)) {
+        if (isSuccess(log) || isSuccess.apply(log)) {
```

Comment:
If a `successFunction` is specified, the default `isSuccess(log)` condition should be ignored.
How about making `successFunction` nullable?
```java
final boolean success;
if (successFunction != null) {
    success = successFunction.test(log);
} else {
    success = isSuccess(log);
}

if (success) {
    ...
} else {
    ...
}
```